server/app/controllers/admin/ProgramAdminManagementController.java

```diff
@@ -85,6 +85,36 @@ public Result update(Http.Request request, long programId) {
     }
   }
 
+  /** Deletes an existing admin email. */
+  @Secure(authorizers = Authorizers.Labels.CIVIFORM_ADMIN)
+  public Result delete(Http.Request request, long programId, String adminEmail) {
+    try {
+      roleService.removeProgramAdmins(programId, ImmutableSet.of(adminEmail));
+      return redirect(routes.AdminProgramController.index());
+    } catch (ProgramNotFoundException e) {
+      return notFound(e.getLocalizedMessage());
+    }
+  }
+
+  /** Adds a new admin email. */
+  @Secure(authorizers = Authorizers.Labels.CIVIFORM_ADMIN)
+  public Result add(Http.Request request, long programId, String adminEmail) {
+    try {
+      Optional<CiviFormError> maybeError =
+          roleService.makeProgramAdmins(programId, ImmutableSet.of(adminEmail));
+
+      if (maybeError.isEmpty()) {
+        return redirect(routes.AdminProgramController.index());
```

Comment:
Same comment here re: redirect location. You can call loadProgram(request, programId, Optional.empty()) instead.